File: 251001/Palitykina/StoryHub/src/main/java/com/example/rest/entity/Story.java
package com.example.rest.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@RequiredArgsConstructor
@Table(name = "tbl_story")
public class Story {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(unique = true, nullable = false)
    private String title;
    private String content;
    private OffsetDateTime created;
    private OffsetDateTime modified;

    @OneToMany(mappedBy = "story", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Message> messages = new ArrayList<>();

    @ManyToMany
    @JoinTable(
            name = "story_label",
            joinColumns = @JoinColumn(name = "storyId"),
            inverseJoinColumns = @JoinColumn(name = "labelId")
    )
    private final List<Label> labels = new ArrayList<>();
}

File: 251002/Belyatskiy/11/DistComp_1/Models/Story.cs
namespace DistComp_1.Models;

public class Article : BaseModel
{
    public string Title { get; set; }
    
    public long UserId { get; set; }
    public User User { get; set; }

    public List<Message> Messages { get; set; } = [];
    
    public string Content { get; set; }
    public DateTime Created { get; set; }
    public DateTime Modified { get; set; }

    public List<Mark> Marks { get; set; } = [];
}